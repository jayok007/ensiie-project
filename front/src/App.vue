<template>
  <v-app id="app">
    <v-toolbar app>
      <v-btn to="/" fab flat color="transparent" icon>
        <img src="./assets/logo.png" alt="logo" height="45" width="45">
      </v-btn>

      <v-toolbar-title>Trouvemonbar.com</v-toolbar-title>

      <v-spacer></v-spacer>

      <v-toolbar-items>
        <template v-if="isAuthenticated">
          <v-menu bottom left offset-y>
            <v-btn icon large slot="activator" color="success">
              <v-icon>account_circle</v-icon>
            </v-btn>
            <v-list>
              <v-list-tile @click="$router.push('/me')">
                <v-list-tile-title>Mon compte</v-list-tile-title>
              </v-list-tile>
              <v-list-tile @click="logout">
                <v-list-tile-title>DÃ©connexion</v-list-tile-title>
              </v-list-tile>
            </v-list>
          </v-menu>
        </template>

        <template v-else>
          <v-btn color="success" flat to="/signup">S'inscrire</v-btn>
          <v-btn color="success" flat to="/signin">Connexion</v-btn>
          </template>
      </v-toolbar-items>
    </v-toolbar>

    <v-content>
      <v-container>
        <v-fade-transition mode="out-in">
          <router-view/>
        </v-fade-transition>
      </v-container>
    </v-content>
  </v-app>
</template>

<script>
export default {
  name: 'App',

  computed: {
    isAuthenticated () { return this.$store.getters.isAuthenticated },
    user () { return this.$store.state.user }
  },

  methods: {
    logout () {
      this.$log.debug('logout', this.user)
      this.$store.dispatch('logout')
    }
  }
}
</script>
