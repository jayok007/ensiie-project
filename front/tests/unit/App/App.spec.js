import { shallowMount } from '@vue/test-utils'
import App from '@/App'

describe('App.vue', () => {
  it('should render properly when not authenticated', () => {
    const wrapper = shallowMount(App, {
      mocks: {
        $store: {
          getters: {
            isAuthenticated: false
          }
        }
      },
      stubs: ['router-link', 'router-view']
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render properly when authenticated', () => {
    const wrapper = shallowMount(App, {
      mocks: {
        $store: {
          getters: {
            isAuthenticated: true
          }
        }
      },
      stubs: ['router-link', 'router-view']
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
