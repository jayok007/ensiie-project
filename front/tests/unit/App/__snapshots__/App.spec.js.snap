// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render properly when authenticated 1`] = `
<vapp-stub id="app">
  <vtoolbar-stub scrollthreshold="300" app="">
    <vbtn-stub activeclass="v-btn--active" fab="true" flat="true" icon="true" tag="button" type="button" to="/" color="transparent"><img src="./assets/img/logo2.png" alt="logo" height="60" width="60"></vbtn-stub>
    <vtoolbartitle-stub class="purple-color">Trouvemonbar.com</vtoolbartitle-stub>
    <vspacer-stub></vspacer-stub>
    <vtoolbaritems-stub>
      <vmenu-stub closeonclick="true" closeoncontentclick="true" maxheight="auto" offsety="true" openonclick="true" origin="top left" transition="v-menu-transition" bottom="" left="">
        <vbtn-stub activeclass="v-btn--active" icon="true" large="true" tag="button" type="button" color="primary">
          <vicon-stub>account_circle</vicon-stub>
        </vbtn-stub>
        <vlist-stub>
          <vlisttile-stub activeclass="primary--text" to="/admin">
            <vlisttiletitle-stub>Administration</vlisttiletitle-stub>
          </vlisttile-stub>
          <vlisttile-stub activeclass="primary--text" to="/feed">
            <vlisttiletitle-stub>Mon feed</vlisttiletitle-stub>
          </vlisttile-stub>
          <vlisttile-stub activeclass="primary--text" to="/me">
            <vlisttiletitle-stub>Mon compte</vlisttiletitle-stub>
          </vlisttile-stub>
          <vlisttile-stub activeclass="primary--text">
            <vlisttiletitle-stub>DÃ©connexion</vlisttiletitle-stub>
          </vlisttile-stub>
        </vlist-stub>
      </vmenu-stub>
    </vtoolbaritems-stub>
  </vtoolbar-stub>
  <vcontent-stub tag="main">
    <vcontainer-stub tag="div">
      <vfadetransition-stub mode="out-in" origin="top center 0">
        <router-view-stub></router-view-stub>
      </vfadetransition-stub>
    </vcontainer-stub>
  </vcontent-stub>
  <snackbar-stub></snackbar-stub>
</vapp-stub>
`;

exports[`App.vue should render properly when not authenticated 1`] = `
<vapp-stub id="app">
  <vtoolbar-stub scrollthreshold="300" app="">
    <vbtn-stub activeclass="v-btn--active" fab="true" flat="true" icon="true" tag="button" type="button" to="/" color="transparent"><img src="./assets/img/logo2.png" alt="logo" height="60" width="60"></vbtn-stub>
    <vtoolbartitle-stub class="purple-color">Trouvemonbar.com</vtoolbartitle-stub>
    <vspacer-stub></vspacer-stub>
    <vtoolbaritems-stub>
      <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" color="primary" to="/signup">S'inscrire</vbtn-stub>
      <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" color="primary" to="/signin">Connexion</vbtn-stub>
    </vtoolbaritems-stub>
  </vtoolbar-stub>
  <vcontent-stub tag="main">
    <vcontainer-stub tag="div">
      <vfadetransition-stub mode="out-in" origin="top center 0">
        <router-view-stub></router-view-stub>
      </vfadetransition-stub>
    </vcontainer-stub>
  </vcontent-stub>
  <snackbar-stub></snackbar-stub>
</vapp-stub>
`;
